app/stores/modelStore.ts

```diff
@@ -0,0 +1,110 @@
+import { NewProject, Project } from '@/types/types';
+import { computed, makeAutoObservable, observable } from 'mobx';
+import customFetch from '../helpers/customFetch';
+
+// Note: Today, just using 1 ModelStore to store all state for all objects for simplicity
+// In the future we may want to split by Object type.
+export class _ModelStore {
+
+    projectsByID: Map<string, Project> = observable.map(new Map());
+
+    constructor() {
+        makeAutoObservable(this);
```

Comment:
This `makeAutoObservable` is somewhat a bit of magic. Basically for any property declared on this class, mobx will detect updates and rerender any components listening on them.

For example, the `projectsByID` property is declared so mobx will track it. If you wanted to add another property/list/object here it would also be tracked